### PR TITLE
feat(security): Implementation for adding registry roles on the fly

### PR DESCRIPTION
**Pull request by jim-wang-intel**
Implement the capability to add registry ACL roles on the fly via environment variable.
Changes on `security-bootstrapper`'s setupRegistryACL sub-command:
 - Add logic to read and parse the environment variable `ADD_REGISTRY_ACL_ROLES` for adding additional registry roles on the fly
 - Add validation logic on the comma-separated list of service key names as role names
 - Add logic to make a unique list of role names combining both ones from configuration and from env. var.
 - Refactor logic to re-set up when running in 2nd time or later

Closes: #3280

Signed-off-by: Jim Wang <yutsung.jim.wang@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?

There is no dynamic way to add registry roles on the fly; only statically from configuration toml.

## What is the new behavior?
Add the feature/capability to allow user to define a list of comma-separated service keys as role names via environment variable `ADD_REGISTRY_ACL_ROLES` to create ACL roles on the fly.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

## New Imports

- [ ] Yes
- [x ] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information
To test this locally, git clone this PR and make build a local version in docker via `make docker` and then one can optionally insert the environment variables `ADD_REGISTRY_ACL_ROLES` in the `consul` service to have the ACL roles created. 
One example:
```yaml
  consul:
  ......
    entrypoint:
    - /edgex-init/consul_wait_install.sh
    environment:
      ADD_REGISTRY_ACL_ROLES: app-service-http-export-secrets, device-virtual
      API_GATEWAY_HOST: kong
...
```

Run the dev version of docker compose: `make run dev` and you should observe the docker logging message containing "successfully created a role...." like the followings:
```
$ docker logs edgex-core-consul
......
level=INFO ts=2021-03-23T22:00:42.647374732Z app=edgex-security-bootstrapper source=aclpolicies.go:141 msg="successfully created a new agent policy with name edgex-service-policy"
level=INFO ts=2021-03-23T22:00:42.649056703Z app=edgex-security-bootstrapper source=command.go:319 msg="Adding role names from environment variable ADD_REGISTRY_ACL_ROLES"
level=INFO ts=2021-03-23T22:00:42.64908756Z app=edgex-security-bootstrapper source=command.go:304 msg="successfully got unique role names, total size=7"
level=INFO ts=2021-03-23T22:00:42.650347073Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [edgex-support-notifications] for secretstore"
level=INFO ts=2021-03-23T22:00:42.651347501Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [edgex-support-scheduler] for secretstore"
level=INFO ts=2021-03-23T22:00:42.652452517Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [edgex-core-command] for secretstore"
level=INFO ts=2021-03-23T22:00:42.653535035Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [app-service-http-export-secrets] for secretstore"
level=INFO ts=2021-03-23T22:00:42.654827517Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [device-virtual] for secretstore"
level=INFO ts=2021-03-23T22:00:42.655944713Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [edgex-core-data] for secretstore"
level=INFO ts=2021-03-23T22:00:42.65696058Z app=edgex-security-bootstrapper source=aclroles.go:129 msg="successfully created a role [edgex-core-metadata] for secretstore"
......

```

Also should observe the similar working messages when it runs 2nd times or later; i.e., docker-compose down and up several time.

### CODE REVIEW

File: internal/security/bootstrapper/command/setupacl/command.go [hunk 1]
```diff
@@ -57,6 +58,9 @@ const (
 	defaultRetryTimeout        = 30 * time.Second
 	emptyLeader                = `""`
 	emptyToken                 = ""
+
+	// environment variable contains a comma separated list of registry role names to be added
+	addRegistryRolesEnvKey = "ADD_REGISTRY_ACL_ROLES"
```
Comment (lenny-intel) [inline internal/security/bootstrapper/command/setupacl/command.go:63]: Please add edgex-doc issue to document this feature.

**Review by lenny-intel — APPROVED** [review]
LGTM

File: internal/security/bootstrapper/command/setupacl/command.go (hunk 1 above)
Comment (jim-wang-intel) [reply internal/security/bootstrapper/command/setupacl/command.go:63]: ok, i'll create one issue

Comment (jim-wang-intel) [reply internal/security/bootstrapper/command/setupacl/command.go:63]: https://github.com/edgexfoundry/edgex-docs/issues/367

**Pull request merged**